index.html

```diff
@@ -0,0 +1,261 @@
+  
+              tbl[rowNum][2] = checked;
+          });
+      }
+      function addDateFieldEvent(row) {
+          row.cells[3].childNodes[0].addEventListener('change', function (event) {
+              var input = event.target || event.srcElement,
+                  cell = input.parentNode,
+                  rowNum = cell.parentNode.rowIndex - 1,
+                  date = input.value;
+  
+              tbl[rowNum][3] = date;
+          });
+      }
+      function addDeleteFieldEvent(row) {
+          row.cells[4].childNodes[0].addEventListener('click', function (event) {
+              var input = event.target || event.srcElement,
+                  cell = input.parentNode,
+                  rowNum = cell.parentNode.rowIndex - 1,
+                  table = document.querySelector("#tblBody");
+  
+              table.deleteRow(rowNum);
+              tbl.splice(rowNum, 1);
+          });
+      }
+  
+      function assignEvents(row) {
+          addStringFieldEvent(row);
+          addNumberFieldEvent(row);
+          addBooleanFieldEvent(row);
+          addDateFieldEvent(row);
+          addDeleteFieldEvent(row);
+      }
+  
+      function addNameCell(nameCell) {
+          var name = document.createElement('input');
+          name.setAttribute('type', 'text');
+          name.setAttribute('class', 'inputField');
+          nameCell.appendChild(name);
+      }
+  
+      function addNumberCell(numberCell) {
+          var number = document.createElement('input');
+          number.setAttribute('type', 'number');
+          number.setAttribute('class', 'inputField');
+          numberCell.appendChild(number);
+      }
+  
+      function addBooleanCell(booleanCell) {
+          var checkbox = document.createElement('input');
+          checkbox.setAttribute('type', 'checkbox');
+          checkbox.setAttribute('class', 'center');
+          checkbox.checked = false;
+          booleanCell.setAttribute('class', 'btn');
+          booleanCell.appendChild(checkbox);
+      }
+  
+      function addDateCell(dateCell) {
+          var date = document.createElement('input');
+          date.setAttribute('type', 'date');
+          date.setAttribute('name', 'DateCell');
+          date.setAttribute('class', 'center');
+          dateCell.setAttribute('class', 'btn');
+          dateCell.appendChild(date);
+      }
+  
+      function addDeleteCell(deleteCell) {
+          var deleteBtn = document.createElement('input');
+          deleteBtn.setAttribute('type', 'button');
+          deleteBtn.setAttribute('name', 'DeleteCell');
+          deleteBtn.setAttribute('value', 'Delete Row');
+          deleteBtn.setAttribute('class', 'center');
+          deleteCell.setAttribute('class', 'btn');
+          deleteCell.appendChild(deleteBtn);
+      }
+     
+  }
+</script>
+
+<table id='myTable'>
+    <thead id='tblHeader'>
+            <th class='name'>
+                <lable class='label'>String</lable>
```

Comment:
What's `<lable>`? I'm not familiar with that element. I bet you meant `<label>`, why `<label>` here?